tools/cjxl_ng_main.cc

```diff
@@ -0,0 +1,604 @@
+
+DEFINE_int64(center_y, -1,
+             // TODO(tfish): Clarify if this is really the comment we want here.
+             "Determines the vertical position of center for the center-first "
+             "group order. The value -1 means 'use the middle of the image', "
+             // TODO(tfish): Clarify if encode.h has an off-by-one in the
+             // upper limit here.
+             "other values 0..(ysize-1) set this to a particular coordinate.");
+
+DEFINE_int64(num_threads, 0,
+             // TODO(tfish): Sync with team about changed meaning of 0 -
+             // was: No multithreaded workers. Is: use default number.
+             "Number of worker threads (0 == use machine default).");
+
+DEFINE_int64(num_reps, 1,  // TODO(tfish): wire this up.
+                           // TODO(tfish): Clarify meaning of this docstring.
+                           // Is this simply for benchmarking?
+             "How many times to compress.");
+
+DEFINE_int32(photon_noise, 0,
+             // TODO(tfish): Discuss docstring change with team.
+             // Also: This now is an int, no longer a float.
+             "Adds noise to the image emulating photographic film noise. "
+             "The higher the given number, the grainier the image will be. "
+             "As an example, a value of 100 gives low noise whereas a value "
+             "of 3200 gives a lot of noise. The default value is 0.");
+
+DEFINE_string(
+    distance, "1.0",  // TODO(tfish): wire this up.
+    "Max. butteraugli distance, lower = higher quality. Range: 0 .. 25.\n"
+    "    0.0 = mathematically lossless. Default for already-lossy input "
+    "(JPEG/GIF).\n"
+    "    1.0 = visually lossless. Default for other input.\n"
+    "    Recommended range: 0.5 .. 3.0.");
+
+DEFINE_int64(target_size, 0,  // TODO(tfish): wire this up.
+             "Aim at file size of N bytes.\n"
+             "    Compresses to 1 % of the target size in ideal conditions.\n"
+             "    Runs the same algorithm as --target_bpp");
+
+DEFINE_string(target_bpp, "0",  // TODO(tfish): wire this up.
+              "Aim at file size that has N bits per pixel.\n"
+              "    Compresses to 1 % of the target BPP in ideal conditions.");
+
+DEFINE_string(
+    quality, "100.0",  // TODO(tfish): wire this up.
+    "Quality setting (is remapped to --distance). Range: -inf .. 100.\n"
+    "    100 = mathematically lossless. Default for already-lossy input "
+    "(JPEG/GIF).\n    Positive quality values roughly match libjpeg "
+    "quality.");
+
+DEFINE_int64(
+    effort, 7,
+    // TODO(tfish): Clarify discrepancy with team:
+    // Documentation says default==squirrel(7) here:
+    // https://libjxl.readthedocs.io/en/latest/api_encoder.html#_CPPv424JxlEncoderFrameSettingId
+    // but enc_params.h has kFalcon=7.
+    "Encoder effort setting. Range: 1 .. 9.\n"
+    "    Default: 7. Higher number is more effort (slower).");
+
+namespace {
+
+// RAII-wraps the C-API encoder.
```

Comment:
This should already exist in encode_cxx.h.

Reply:
decided to use `encode_cxx.h` and `thread_parallel_runner_cxx.h`